pkg/auth/auth.go

```diff
@@ -0,0 +1,305 @@
+package auth
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"github.com/klothoplatform/klotho/pkg/closenicely"
+	"github.com/pkg/errors"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/klothoplatform/klotho/pkg/cli_config"
+	"github.com/pkg/browser"
+	"go.uber.org/zap"
+)
+
+var (
+	authUrlBase          = EnvVar("KLOTHO_AUTH_BASE").GetOr(`http://klotho-auth-service-alb-e22c092-466389525.us-east-1.elb.amazonaws.com`)
+	pemUrl               = EnvVar("KLOTHO_AUTH_PEM").GetOr(`https://klotho.us.auth0.com/pem`)
+	ErrNoCredentialsFile = errors.New("no local credentials file")
+	ErrEmailUnverified   = errors.New("login email hasn't been verified")
+)
+
+type LoginResponse struct {
+	Url   string
+	State string
+}
+
+type Authorizer interface {
+	// Authorize tries to authorize the user. The KlothoClaims it returns may be nil, even if the authentication
+	// succeeds. Conversely, if the KlothoClaims is non-nil, it is valid even if the error is also non-nil; you can use
+	// those claims provisionally (and specifically, in analytics) even if the error is non-nil, indicating failed
+	// authentication.
+	Authorize() (*KlothoClaims, error)
+}
+
+func DefaultIfNil(auth Authorizer) Authorizer {
+	if auth == nil {
+		return standardAuthorizer{}
+	}
+	return auth
+}
+
+type standardAuthorizer struct{}
+
+func (s standardAuthorizer) Authorize() (*KlothoClaims, error) {
+	return Authorize()
+}
```

Comment:
I'd prefer to keep this as-is for now, since (a) it works, (b) refactoring it later is unlikely to be more work than refactoring it now, and (c) we currently don't have a need for multiple authorizers.

Reply:
(a) sure

(b) makes sense

(c) I'd argue that the local implementation is already multiple authorizers. If not, then there's no need for a struct or interface - just call the global methods from klothomain.